anyway/infographics_utils.py

```diff
@@ -949,25 +949,90 @@ class MotorcycleAccidentsVsAllAccidentsWidget(Widget):
         self.rank = 20
-        self.text = {"title": "תאונות אופנועים קשות וקטלניות בכביש 20 בהשוואה לכל הארץ"}
+        self.road_number = request_params.location_info["road1"]
+        self.text = {"title": f"תאונות אופנועים קשות וקטלניות בכביש {int(self.road_number)} בהשוואה לכל הארץ"}
 
     # noinspection PyMethodMayBeStatic
     def is_in_cache(self) -> bool:
         return False
 
     def generate_items(self) -> None:
         self.items = (
-            MotorcycleAccidentsVsAllAccidentsWidget.motorcycle_accidents_vs_all_accidents_mock_data()
+            MotorcycleAccidentsVsAllAccidentsWidget.motorcycle_accidents_vs_all_accidents(
+                self.request_params.start_time, self.request_params.end_time, self.road_number)
         )
 
     @staticmethod
-    def motorcycle_accidents_vs_all_accidents_mock_data():  # Temporary for Frontend
+    def motorcycle_accidents_vs_all_accidents(start_time, end_time, road_number):
+        location_label = "location"
+        location_all = "כל הארץ"
+        location_road = f'כביש {int(road_number)}'
+        case_location = case([
+            ((InvolvedMarkerView.road1 == road_number) | (InvolvedMarkerView.road2 == road_number),
+             location_road)
+        ],
+            else_=literal_column(f"'{location_all}'")
+        ).label(location_label)
+
+        vehicle_label = "vehicle"
+        vehicle_other = "אחר"
+        vehicle_motorcycle = "אופנוע"
+        case_vehicle = case([(
+            InvolvedMarkerView.involve_vehicle_type.in_(BE_CONST.MOTORCYCLE_VEHICLE_TYPES),
+            literal_column(f"'{vehicle_motorcycle}'")
+        )],
+            else_=literal_column(f"'{vehicle_other}'")
+        ).label(vehicle_label)
+
+        query = get_query(table_obj=InvolvedMarkerView, filters={}, start_time=start_time,
+                          end_time=end_time)
+
+        query_columns = [InvolvedMarkerView.accident_id, InvolvedMarkerView.provider_and_id,
+                         InvolvedMarkerView.road_type, InvolvedMarkerView.accident_severity,
+                         InvolvedMarkerView.provider_code, InvolvedMarkerView.involve_vehicle_type,
+                         InvolvedMarkerView.road1, InvolvedMarkerView.road2]
+
+        num_accidents_label = "num_of_accidents"
+        query = (query.with_entities(*query_columns, case_location, case_vehicle,
+                                     func.count(distinct(InvolvedMarkerView.provider_and_id))
+                                     .label(num_accidents_label))
+                 .filter(InvolvedMarkerView.road_type.in_(BE_CONST.NON_CITY_ROAD_TYPES))
+                 .filter(InvolvedMarkerView.accident_severity.in_([BE_CONST.ACCIDENT_SEVERITY_DEADLY,
+                                                                   BE_CONST.ACCIDENT_SEVERITY_SEVERE]))
+                 .group_by(*query_columns)
+                 .order_by(desc(InvolvedMarkerView.provider_code))
```

Comment:
This code does not match the original query code( `GROUP BY 1,2 ORDER BY 3 DESC `), What is it suppose to do?

Reply:
@BarVolunteering is right - query_columns contain, for example, accident_id and provider_and_id - do we want to group by all columns in query_columns?